### PR TITLE
Feature/unique title validation playlistcreation

### PR DESCRIPTION
**Pull request by goody153**
## Ticket 
  - [playlist name should be unique for creator](https://trello.com/c/BSJIjRdI)

## Summary
  - Changed save override on Forms
  - Removed UpdateSongForm
  - Unique playlist title

### CODE REVIEW

File: playlist/models.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 class Playlist(models.Model):
     """Playlist
     """
-    title = models.CharField(max_length=255)
+    title = models.CharField(max_length=255, unique=True)
```
Comment (mcandidier) [inline playlist/models.py]: this will be unique to all entries

File: playlist/forms.py [hunk 2]
```diff
@@ -28,29 +42,24 @@ def __init__(self, *args, **kwargs):
...
-    """
-
-    class Meta:
-        model = Song
-        fields = ['title', 'link']
\ No newline at end of file
+    def save(self, commit=True):
+        """ Song creation needs user and playlist 
+        """
+        instance = super(SongForm, self).save(commit=False)
```
Comment (mcandidier) [inline playlist/forms.py]: rename instance to `song` to avoid confusion to self.instance

File: playlist/forms.py [hunk 3]
```diff
@@ -28,29 +42,24 @@ def __init__(self, *args, **kwargs):
...
+        #checks if self.user and self.playlist exists to determine if it's a creation process
+        if self.user and self.playlist:
```
Comment (mcandidier) [inline playlist/forms.py]: use ternary operator instead of if else statement.
myvar = `foo` if foo else `bar`

**Comment by goody153** [comment]
Made the changes Sir but i used clean for the validation error of unique name.

**Pull request merged**